basic-configuration.md

```diff
@@ -1,121 +1,128 @@
-categories:
-
-1. Go to **Stores** > **Configuration**.
-2. In the left-hand menu, click on **Catalog** > **Catalog**.
-3. Expand the **Category Top Navigation** section.
-4. Enable the **Enable Category Top Navigation Menu** option.
-5. Click on **Save Config**.
-
-## Payment Configuration
-
-Payment configuration settings in Magento 2 allow you to set up and manage the payment methods available to your
-customers.
-
-To access the payment configuration, follow these steps:
-
-1. Log in to the Magento 2 admin panel.
-2. Navigate to **Stores** > **Configuration**.
-
-### Example: Enabling PayPal Express Checkout
-
-PayPal Express Checkout is a popular payment method. To enable PayPal Express Checkout:
-
-1. Go to **Stores** > **Configuration**.
-2. In the left-hand menu, click on **Sales** > **Payment Methods**.
-3. Expand the **PayPal Express Checkout** section.
-4. Set the **Enabled** option to **Yes**.
-5. Enter your PayPal API credentials in the appropriate fields.
-6. Click on **Save Config**.
-
-## Shipping Configuration
-
-Shipping configuration settings in Magento 2 allow you to manage how shipping is calculated and handled in your store.
-
-To access the shipping configuration, follow these steps:
-
-1. Log in to the Magento 2 admin panel.
-2. Navigate to **Stores** > **Configuration**.
-
-### Example: Configuring Flat Rate Shipping
-
-Flat rate shipping charges a fixed rate for shipping regardless of the order's weight or location. To configure flat
-rate shipping:
-
-1. Go to **Stores** > **Configuration**.
-2. In the left-hand menu, click on **Sales** > **Shipping Methods**.
-3. Expand the **Flat Rate** section.
-4. Set the **Enabled** option to **Yes**.
-5. Configure the **Title**, **Method Name**, **Price**, and other relevant fields.
-6. Click on **Save Config**.
-
-## Conclusion
-
-In this documentation, we have covered the basic configuration settings in Magento 2, including system, store, catalog,
-payment, and shipping configurations. By following the provided instructions and using the code snippets and examples,
-you can easily configure your Magento 2 store to meet your specific requirements. For more advanced configuration
-options, refer to the official Magento 2 documentation or consult with a Magento developer.
+## Introduction
+
+Magento 2, a versatile eCommerce platform, offers extensive configurability to meet the needs of developers. This
+document discusses the fundamental configuration in Magento 2, focusing on `app/etc/env.php`, `app/etc/config.php`,
+`core_config_data` table, `config.xml`, and managing configuration programmatically.
+
+## Environment Configuration in `app/etc/env.php`
+
+The `app/etc/env.php` file is pivotal for environment-specific configurations. It stores settings such as database
+connection information, backend front name, cryptographic key, and more.
+
+### Database Connection
```

Comment:
We could description the replication setup here as well. Maybe as separate chapter?

This is an example config:

```
return array (
      //...
      'db' =>
         array (
            'connection' =>
               array (
                  'indexer' =>
                     array (
                        'host' => 'default-master-host',
                        'dbname' => 'magento',
                        'username' => 'magento',
                        'password' => 'magento',
                        'active' => '1',
                        'persistent' => NULL,
                     ),
                  'default' =>
                     array (
                        'host' => 'default-master-host',
                        'dbname' => 'magento',
                        'username' => 'magento',
                        'password' => 'magento',
                        'active' => '1',
                     ),
                  'checkout' =>
                     array (
                        'host' => 'checkout-master-host',
                        'dbname' => 'checkout',
                        'username' => 'magento',
                        'password' => 'magento',
                        'model' => 'mysql4',
                        'engine' => 'innodb',
                        'initStatements' => 'SET NAMES utf8;',
                        'active' => '1',
                     ),
                  'sales' =>
                     array (
                        'host' => 'sales-master-host',
                        'dbname' => 'sales',
                        'username' => 'magento',
                        'password' => 'magento',
                        'model' => 'mysql4',
                        'engine' => 'innodb',
                        'initStatements' => 'SET NAMES utf8;',
                        'active' => '1',
                     ),
               ),
            'slave_connection' =>
               array (
                  'default' =>
                     array (
                        'host' => 'default-slave-host',
                        'dbname' => 'magento',
                        'username' => 'read_only',
                        'password' => 'password',
                        'active' => '1',
                     ),
                  'checkout' =>
                     array (
                        'host' => 'checkout-slave-host',
                        'dbname' => 'checkout',
                        'username' => 'read_only',
                        'password' => 'password',
                        'model' => 'mysql4',
                        'engine' => 'innodb',
                        'initStatements' => 'SET NAMES utf8;',
                        'active' => '1',
                     ),
                  'sales' =>
                     array (
                        'host' => 'sales-slave-host',
                        'dbname' => 'sales',
                        'username' => 'read_only',
                        'password' => 'password',
                        'model' => 'mysql4',
                        'engine' => 'innodb',
                        'initStatements' => 'SET NAMES utf8;',
                        'active' => '1',
                     ),
                  ),
               'table_prefix' => '',
   ),
   //.......
```

Reply:
That's definitely worth its own chapter, yes

Reply:
@DavidLambauer would you like to add the chapter to this .md file or to another one? In the latter, can I consider this thread as closed in order to move on with the merge? Thank you guys!